packages/debug/src/browser/view/debug-configuration-select.tsx

```diff
@@ -0,0 +1,227 @@
+            onBlur={this.refreshDebugConfigurations}
+        >
+            {this.renderOptions()}
+        </select>;
+    }
+
+    protected get currentValue(): string {
+        const { current } = this.manager;
+        return current ? InternalDebugSessionOptions.toValue(current) : DebugConfigurationSelect.NO_CONFIGURATION;
+    }
+
+    protected readonly setCurrentConfiguration = (event: React.ChangeEvent<HTMLSelectElement>) => {
+        const value = event.currentTarget.value;
+        if (value === DebugConfigurationSelect.ADD_CONFIGURATION) {
+            this.manager.addConfiguration();
+        } else if (value.startsWith(DebugConfigurationSelect.PICK)) {
+            const providerType = this.parsePickValue(value);
+            this.selectDynamicConfigFromQuickPick(providerType);
+        } else {
+            const [name, type, request, workspaceFolderUri, providerType] = InternalDebugSessionOptions.parseValue(value);
+            this.manager.current =
+                this.manager.find
+                    (
+                        { name, type, request },
+                        workspaceFolderUri,
+                        providerType === 'undefined' ? undefined : providerType
+                    );
+        }
+    };
+
+    protected toPickValue(providerType: string): string {
+        return DebugConfigurationSelect.PICK + providerType;
+    }
+
+    protected parsePickValue(value: string): string {
+        return value.slice(DebugConfigurationSelect.PICK.length);
+    }
+
+    protected async selectDynamicConfigFromQuickPick(providerType: string): Promise<void> {
+        const configurationsOfProviderType =
+            (await this.manager.provideDynamicDebugConfigurations()).find(entry => entry.type === providerType);
+        if (!configurationsOfProviderType) {
+            return;
+        }
+        const { configurations } = configurationsOfProviderType;
+
+        const picks: DynamicPickItem[] = [];
+        for (const configuration of configurations) {
+            picks.push({
+                label: configuration.name,
+                configurationType: configuration.type,
+                request: configuration.request,
+                providerType
+            });
+        }
+
+        if (picks.length === 0) {
+            return;
+        }
+
+        const quickPick = this.quickInputService.createQuickPick<DynamicPickItem>();
+        quickPick.items = picks;
+        quickPick.placeholder = nls.localizeByDefault('Select Launch Configuration');
+        quickPick.show();
+
+        const selected: DynamicPickItem | undefined = await new Promise(resolve => {
+            // If the user presses `Escape` then `quickPick.onDidAccept` will fire
+            // and `quickPick.activeItems` will be empty.
+            quickPick.onDidAccept(() => {
+                resolve(quickPick.activeItems[0]);
+            });
+        });
+
+        quickPick.dispose();
```

Comment:
I believe `this.quickInputService.showQuickPick` can replace all of this code.
It accepts the items, and supports configuration options such as `placeholder`.

Reply:
Done! Thanks !